### DISCUSSION TITLE
Problem check updates while and after installation MikTex

### DISCUSSION THREAD

**Discussion by commelect** [MiKTeX/miktex#1512] [General]
I have a problem after installing Miktex. I couldn't update packages which affected my work on Texmaker for missing packages
I have tried installation servel times on many PCs.
I tried two set up file basic-miktex-23.10-x64 and basic-miktex-24.10-x64. and in every time I got the same error :Timeout reached
Below copy of the error message:
------------------------------------
GENERAL MIKTEX INFORMATION
ReportDate: 2024-07-25 11:03:04
CurrentVersion: 23.10
SetupDate: 2024-07-25 10:59:00
SetupVersion: 23.10
Configuration: Regular
GitInfo: 3f4a6a7 / 2023-10-04 18:18:19
OS: Windows 10.0.17763
SharedSetup: yes
LinkTargetDirectory: C:\Latex\MiKTeX\miktex\bin\x64
PathOkay: no
LastUpdateCheckAdmin: not yet
LastUpdateAdmin: not yet
LastUpdateDbAdmin: not yet
SystemAdmin: yes
RootPrivileges: yes
AdminMode: yes
Root0: C:\ProgramData\MiKTeX
Root1: C:\Latex\MiKTeX
CommonInstall: C:\Latex\MiKTeX
CommonConfig: C:\ProgramData\MiKTeX
CommonData: C:\ProgramData\MiKTeX
Invokers: non-existing/cmd

The following issues were detected:
  1: major issue: So far, no MiKTeX administrator has checked for updates.

ERROR DETAILS
Program: miktex-console
Source: Libraries\MiKTeX\PackageManager\CurlWebSession.cpp:247
Error: Timeout was reached
Details: 
  code: 28
  url: https://ctan.yazd.ac.ir/systems/win32/miktex/tm/packages/miktex-zzdb3-2.9.tar.lzma

**Comment by aglobalphenomenon** [comment]
I know this is late, but in case someone faces the same issue, I managed to stitch a simple solution in my case (on Windows).

When updating packages, MiKTeX’s package manager (`mpm`) automatically selects the closest mirror. Sometimes the closest mirror (e.g. one at https://ctan.yazd.ac.ir/) may be hosted in a region (like Iran) where domains are blocked by your region, causing connection timeouts (error code 28).

1. **Manually Set an Accessible Repository**  
   - **Find a Mirror:**  
     Visit [[MiKTeX Repositories](https://miktex.org/pkg/repositories)](https://miktex.org/pkg/repositories) and choose a URL from a country with no geo-blocking issues.
   - **Configure via Command Line:**  
     Open a command prompt and run the following (run elevated/as admin if needed):
     ```
     mpm --set-repository="https://<chosen-repository>/ctan/systems/win32/miktex/tm/packages/"
     ```
     If running in administrator mode (using MiKTeX Console Admin), add the `--admin` flag and use:
     ```
     mpm --admin --set-repository="https://<chosen-repository>/ctan/systems/win32/miktex/tm/packages/"
     ```
     Example :
     ```
     mpm --admin --set-repository="https://mirrors.nxthost.com/ctan/systems/win32/miktex/tm/packages/"
     ```

2. **Use a VPN**  
   Otherwise, instead of all of that, you can use a VPN to temporarily bypass the geo-blocking to reach a different default mirror.
